### PR TITLE
Added bcrypt support for htpasswd

### PR DESCRIPTION
**Pull request by 0x78f1935**
**Feature:**
**htpasswd:**

The following has been addressed in the PR:

*  There is a related issue?
        Probably, there was no support for bcrypt
*  Unit or Functional tests are included in the PR
        Unfortunately I'm unable to run this locally because my IQ is running on windows

**Description:**
Hopefully this will be added to the docker image.
Please let me know if I can do anything else!

Test case could be: `admin:$2y$05$1oQhGWJvMyR6ne24SWNKW.K926uQKfiIdWpZOy1hsd8hX0rLhOYwm`
pwd: admin

### CODE REVIEW

**Review by juanpicado — CHANGES_REQUESTED** [review]
The lint is broken, check the logs. Run `yarn lint` in your branch.

**Comment by juanpicado** [comment]
1. You need to push the updated `yarn.lock`, just ran `yarn` locally.

**Comment by juanpicado** [comment]
The version 5 does not support Node 8. https://www.npmjs.com/package/bcrypt#version-compatibility

Furthermore, this library uses C++ dependencies https://github.com/kelektiv/node.bcrypt.js/blob/master/package.json#L32 which force to compiles on install, I'd like do not include that, maybe try something pure JS liek https://github.com/shaneGirish/bcrypt-nodejs or any other option. At Verdaccio 5 I'd like create an standalone version and to achive that I need 0 dependencies related with https://github.com/nodejs/node-gyp

**Comment by 0x78f1935** [comment]
Gotcha, thanks for your review!, I see what I can do and apply those changes asap.

File: yarn.lock [hunk 1]
```diff
@@ -9470,6 +9480,22 @@ node-notifier@^7.0.0:
     uuid "^7.0.3"
     which "^2.0.2"
 
+node-pre-gyp@0.15.0:
```
Comment (juanpicado) [inline yarn.lock:9483]: I think this lock file does not match, why this have changed if deps didnt'?

**Pull request closed without merging**